src/v/storage/types.h

```diff
@@ -467,19 +467,71 @@ struct usage {
     }
 };
 
+/*
+ * disk usage targets
+ *
+ * min_capacity: minimum amount of storage capacity needed.
+ * min_capacity_wanted: minimum amount needed to meet policy requirements.
+ *
+ * The minimum capacity is intended to capture the minimum amount of disk space
+ * needed for the basic functionality. At a high-level this is expressed as a
+ * minimum number of segments per partition. Formally it is the sum of S *
+ * log.max_segment_size() for each managed log, where S is the value of the
+ * configuration option storage_reserve_min_segments specifying the minimum
+ * number of segments for which space should be reserved.
+ *
+ * The minimum capacity wanted is an estimate of the amount of storage capacity
+ * needed to meet various configurable targets. This value is derived from
+ * multiple sources and policies:
+ *
+ *    * size-based retention: the amount of space needed to meet size-based
+ *    local retention policy, rounded up to the nearest segment size.
+ *
+ *    * time-based retention: attempts to extrapolate the capacity requirements
+ *    by examining recently written data and the apparent rate at which it has
+ *    been written.
+ *
+ *    * compaction: compacted topics are kept whole on local storage (ie not
+ *    subject to truncation due to local retention policies). for compact,delete
+ *    topic the retention policy (not local retention) is used to express how
+ *    much capacity is wanted. for a compact-only topic `2 * current size` is
+ *    reported.
+ */
+struct usage_target {
+    size_t min_capacity{0};
+    size_t min_capacity_wanted{0};
+
+    friend usage_target operator+(usage_target lhs, const usage_target& rhs) {
```

Comment:
nit: is `friend` required here, or can this just be a simple operator overload, as both fields in `usage_target` are public?

Reply:
@abhijat you are right friend is not needed. It's a habit to do that in the tree because I believe there is some sort of compilation time benefit. Never really dug into the details.